src/hotspot/share/oops/klass.hpp

```diff
@@ -652,10 +652,7 @@ class Klass : public Metadata {
   // Note: the prototype header is always set up to be at least the
   // prototype markWord. If biased locking is enabled it may further be
   // biasable and have an epoch.
-  markWord prototype_header() const      {
-    assert(UseCompactObjectHeaders, "only use with compact object headers");
-    return _prototype_header;
-  }
+  markWord prototype_header() const      { return _prototype_header; }
```

Comment:
Right. This is a change that should be done in a separate PR. Thing is that in 17 we could call this method if anybody enabled +UseBiasedLocking, and the assert would fail.